src/compiler/scala/tools/nsc/symtab/SymbolLoaders.scala

```diff
@@ -309,7 +313,7 @@ abstract class SymbolLoaders {
       // errors. More concretely, the classfile parser calls "sym.companionModule", which calls
       // "isModuleNotMethod" on the companion. After refchecks, this method forces the info, which
       // may run the classfile parser. This produces the error.
-      enteringPhase(phaseBeforeRefchecks)(classfileParser.parse(classfile, root))
+      enteringPhase(phaseBeforeRefchecks)(classfileParser.parse(classfile, clazz, module))
```

Comment:
The comment seems outdated, the classfile parser doesn't call `companionModule` anymore. Maybe there's no need to call `enteringPhase` either?